src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -179,7 +179,7 @@ namespace cryptonote
             LOG_PRINT_L1("Failed to parse transaction from blob, bad bulletproofs_plus L size in tx " << get_transaction_hash(tx));
             return false;
           }
-          const size_t max_outputs = 1 << (rv.p.bulletproofs_plus[0].L.size() - 6);
+          const size_t max_outputs = rct::n_bulletproof_plus_max_amounts(rv.p.bulletproofs_plus[0]);
```

Comment:
This must go below the line `rv.p.bulletproofs_plus[0].V[i] = rv.outPk[i].mask;`. The precondition of `n_bulletproof_plus_max_amounts()` is that the bp+ passed in is fully defined.

Reply:
Maybe I'm missing your point, but if you do that, you'll get the max_output check after the resize, which means you can make it allocate a large amount of memory. And AFAICT n_bulletproof_plus_max_amounts does not need V.